dags/ods/kktix_ticket_orders/udfs/kktix_transformer.py

```diff
@@ -0,0 +1,17 @@
+import hashlib
+from typing import Dict, List
+
+
+def transform(event_raw_data_array: List) -> List[Dict]:
+    """
+    de-identify user's email in this block!
+    """
+    for event in event_raw_data_array:
+        for attendee_info in event["attendee_infos"]:
+            # search string contains personal information and it's unstructured. Therefore just drop it!
+            del attendee_info["search_string"]
+            for index, (key, value) in enumerate(attendee_info["data"]):
+                if key in {"聯絡人 姓名", "聯絡人 Email", "聯絡人 手機"}:
```

Comment:
@GeekAngus I forgot to point it out, you should keep the raw and sensitive data that you need in memory, and then kick off the subsequent `loader` pipeline~